src/modules/launcher/Plugins/Microsoft.Plugin.Uri.UnitTests/UriHelper/ExtendedUriParserTests.cs

```diff
@@ -11,87 +11,122 @@ namespace Microsoft.Plugin.Uri.UnitTests.UriHelper
-        [DataRow("ms-settings:powersleep", true, "ms-settings:powersleep", false)]
-        [DataRow("microsoft-edge:http://google.com", true, "microsoft-edge:http://google.com", false)]
-        [DataRow("microsoft-edge:https://google.com", true, "microsoft-edge:https://google.com", false)]
-        [DataRow("microsoft-edge:google.com", true, "microsoft-edge:google.com", false)]
-        [DataRow("microsoft-edge:google.com/", true, "microsoft-edge:google.com/", false)]
-        [DataRow("microsoft-edge:https://google.com/", true, "microsoft-edge:https://google.com/", false)]
-        [DataRow("ftp://user:password@localhost:8080", true, "ftp://user:password@localhost:8080/", false)]
-        [DataRow("ftp://user:password@localhost:8080/", true, "ftp://user:password@localhost:8080/", false)]
-        [DataRow("ftp://user:password@google.com", true, "ftp://user:password@google.com/", false)]
-        [DataRow("ftp://user:password@google.com:2121", true, "ftp://user:password@google.com:2121/", false)]
-        [DataRow("ftp://user:password@1.1.1.1", true, "ftp://user:password@1.1.1.1/", false)]
-        [DataRow("ftp://user:password@1.1.1.1:2121", true, "ftp://user:password@1.1.1.1:2121/", false)]
-        [DataRow("^:", false, null, false)]
+        // Case where `domain:port`, as specified in issue #14260
+        // Assumption: Only domain with dot is accepted
+        [DataRow("example.com:80", true, "http://example.com/", true, null)]
+        [DataRow("example.com:80/test", true, "http://example.com/test", true, null)]
+        [DataRow("example.com:80/126", true, "http://example.com/126", true, null)]
+        [DataRow("example.com:443", true, "https://example.com/", true, null)]
+        [DataRow("example.com:443/test", true, "https://example.com/test", true, null)]
+        [DataRow("example.com:443/126", true, "https://example.com/126", true, null)]
+        [DataRow("google.com:91", true, "https://google.com:91/", true, null)]
+        [DataRow("google.com:91/test", true, "https://google.com:91/test", true, null)]
+        [DataRow("google.com:91/126", true, "https://google.com:91/126", true, null)]
+        [DataRow("test.test.test.test:952", true, "https://test.test.test.test:952/", true, null)]
+        [DataRow("test.test.test.test:952/test", true, "https://test.test.test.test:952/test", true, null)]
+        [DataRow("test.test.test.test:952/126", true, "https://test.test.test.test:952/126", true, null)]
 
-        public void TryParseCanParseHostName(string query, bool expectedSuccess, string expectedResult, bool expectedIsWebUri)
+        // Following cases can be both interpreted as schema:path and domain:port
+        [DataRow("tel:411", true, "https://tel:411/", true, "tel:411")]
+        [DataRow("tel:411/test", true, "https://tel:411/test", true, "tel:411/test")]
+        [DataRow("tel:411/126", true, "https://tel:411/126", true, "tel:411/126")]
+        [DataRow("local:1234", true, "https://local:1234/", true, "local:1234")]
```

Comment:
What is local for a target?

We should add localhost with the following tests. Do you agree?
- `localhost`, `localhost:1234`, `http://localhost`, `http://localhost:1234`, `https://localhost`, `https://localhost:1234`
- `127.0.0.1`, `127.0.0.1:1234`, `http://127.0.0.1`, `http://127.0.0.1:1234`, `https://127.0.0.1`, `https://127.0.0.1:1234`
- `[::1]`, `[::1]:1234`, `http://[::1]`, `http://[::1]:1234`, `https://[::1]`, `https://[::1]:1234`

Reply:
Sure, sounds good to me, I'll add them

Reply:
I saw some localhost check in lines before and maybe we can check for a subpage on the local host too (`localhost:1234/test`).

Reply:
I'm moving all of them into one chuck for easier management

Reply:
Also removing local as target because it's redundant as a test